[iOS] Instagram story header lays out under status bar on the web and can’t be tapped

jernoble opened the pull request:
#### e9d2d1449a98d1e47abca2829f576cbb19e7d041
<pre>
[iOS] Instagram story header lays out under status bar on the web and can’t be tapped
<a href="https://bugs.webkit.org/show_bug.cgi?id=267893">https://bugs.webkit.org/show_bug.cgi?id=267893</a>
<a href="https://rdar.apple.com/121014613">rdar://121014613</a>

Reviewed by Aditya Keerthi and Eric Carlson.

Instagram adopts `viewport-fit=cover`, so fullscreen content will intentionally flow under the
unsafe areas in fullscreen. Adding a quirk for Instagram.com to disable the Fullscreen API
until this behavior is fixed server-side.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldDisableElementFullscreenQuirk const):

Canonical link: <a href="https://commits.webkit.org/273409@main">https://commits.webkit.org/273409@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ec685400e36cd00342c510387f7c760bb4902fcc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35360 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14286 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/37474 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38097 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/31889 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/36557 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/16657 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11331 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/30753 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/35910 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12068 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31488 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10595 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/10633 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/31604 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39343 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32129 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/31936 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36612 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10789 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8695 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/34646 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12545 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8086 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11313 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/11602 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -410,8 +410,10 @@ bool Quirks::shouldDisableElementFullscreenQuirk() const
+    // (Ref: rdar://121014613)
     if (!m_shouldDisableElementFullscreen)
-        m_shouldDisableElementFullscreen = isDomain("vimeo.com"_s);
+        m_shouldDisableElementFullscreen = isDomain("vimeo.com"_s) || isDomain("instagram.com");
```
Is the quirk really needed for iPad?

pxlcoder replied:
```suggestion
        m_shouldDisableElementFullscreen = isDomain("vimeo.com"_s) || isDomain("instagram.com"_s);
```

jernoble committed:
[iOS] Instagram story header lays out under status bar on the web and can’t be tapped
https://bugs.webkit.org/show_bug.cgi?id=267893
rdar://121014613

Reviewed by Aditya Keerthi and Eric Carlson.

Instagram adopts `viewport-fit=cover`, so fullscreen content will intentionally flow under the
unsafe areas in fullscreen. Adding a quirk for Instagram.com to disable the Fullscreen API
until this behavior is fixed server-side.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldDisableElementFullscreenQuirk const):

Canonical link: https://commits.webkit.org/273409@main
